ui-tests/src/test/resources/features/integrations/odata.feature

```diff
@@ -0,0 +1,164 @@
+    Then check "Next" button is "Disabled"
+    And fill in periodic query input with "SELECT * FROM contact WHERE first_name = 'Jianathan'" value
+    And fill in period input with "1" value
+    And select "Hours" from sql dropdown
+    And click on the "Next" button
+    And check that position of connection to fill is "Finish"
+
+    When select the "OData" connection
+    And select "Create" integration action
+    # check button to make sure correct page is loaded
+    Then check visibility of the "Done" button
+    When select "Products" from "resourcePath" dropdown
+    And click on the "Done" button
+
+    When add integration step on position "0"
+    And select "Data Mapper" integration step
+    And open data mapper collection mappings
+    And create data mapper mappings
+      | first_name | Name        |
+      | last_name  | Description |
+    And click on the "Done" button
+
+    Then check visibility of page "add to integration"
+    When click on the "Save" button
+    And set integration name "OData_Create"
+    And publish integration
+    Then Integration "OData_Create" is present in integrations list
+    And wait until integration "OData_Create" gets into "Running" state
+    When sleep for "10000" ms
+
+    Then validate that OData service contains entity with "Name":"Jianathan" property:value pair in "Products" collection
+
+  #
+  # 3. Test for Delete operation
+  #
+  @gh-4889
+  @reproducer
+  @integrations-odata-delete
+  Scenario: Delete entity in OData service
+
+    When click on the "Create Integration" button to create a new integration
+    Then check visibility of visual integration editor
+    And check that position of connection to fill is "Start"
+
+    When selects the "OData" connection
+    And select "Read" integration action
+    Then check visibility of the "Done" button
+    When fill in values by element ID
+      | resourcePath | Products |
+      | keyPredicate | 1        |
+      | delay        | 10       |
+      | select-delay | Minutes  |
+    And click on the "Done" button
+    Then check that position of connection to fill is "Finish"
+
+    When selects the "OData" connection
+    And select "Delete" integration action
+    Then check visibility of the "Next" button
+    When select "Products" from "resourcePath" dropdown
+    And click on the "Next" button
+
+    When add integration step on position "0"
+    And select "Data Mapper" integration step
+    And create data mapper mappings
+      | ID | keyPredicate |
+    And click on the "Done" button
+
+    Then check visibility of page "add to integration"
+    When click on the "Save" button
+    And set integration name "OData_Delete"
+    And publish integration
+    Then Integration "OData_Delete" is present in integrations list
+    And wait until integration "OData_Delete" gets into "Running" state
+    When sleep for "10000" ms
+
+    Then check that entity "1" is not present in "Products" collection on OData service
```

Comment:
I don't see the `Product(1)` entity being created in the test, is the odata service somehow pre-populated with this sample data? I'd probably like to see a check that the entity exists in the service before attempting to delete it, so that we don't get false positives in case the entity isn't there from the start

Reply:
As mentioned above, it is prepopulated. But I agree it would be better to insert the entity in tests so it is obvious what is being deleted. Will redo that.